apps/openchallenges/edam-etl/src/main.py

```diff
@@ -1,16 +1,107 @@
-"""Extract, transform and load EDAM concepts"""
-
+import pandas as pd
+import requests
 from os import getenv
+from typing import Optional
 
 # Get config from the environment variables
-oc_db_url = getenv("OC_DB_URL")
-print(f"OC DB URL: {oc_db_url}")
 
-# TODO Download EDAM concepts from GitHub or S3 bucket (CSV file)
-print("[TODO] Download EDAM concepts from GitHub or S3 bucket (CSV file)")
+OC_DB_URL = getenv("OC_DB_URL")
+VERSION = getenv("VERSION")
+print(f"EDAM Version: {VERSION}")
+print(f"OC DB URL: {OC_DB_URL}")
+
+
+def download_edam_csv(url: str, version: str) -> Optional[bool]:
+    """Download EDAM concepts from GitHub or S3 bucket (CSV file)"""
+    print("Downloading the EDAM concepts from GitHub (CSV file)...")
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an exception for bad response
+        with open(f"EDAM_{version}.csv", "wb") as f:
+            f.write(response.content)
+        print("EDAM concepts downloaded successfully.")
+        return True
+    except requests.RequestException as e:
+        print(f"Error downloading EDAM concepts: {e}")
+        return None
+
+
+def transform_to_dataframe(version: str) -> pd.DataFrame | None:
+    """Transform the CSV to a DataFrame with indices starting from 1"""
+    print("Processing the EDAM concepts...")
+    try:
+        df = (
+            pd.read_csv(f"EDAM_{version}.csv", usecols=["Class ID", "Preferred Label"])
+            .rename(
+                columns={"Class ID": "class_id", "Preferred Label": "preferred_label"}
+            )
+            .reset_index(drop=True)
+            .reset_index(drop=False)
+            .rename(columns={"index": "id"})
+        )
+        print("EDAM concepts processed successfully.")
+        return df
+    except FileNotFoundError as e:
+        print(f"Error processing EDAM concepts: {e}")
+        return None
+
+
+def count_occurrences(identifier_pattern: str, df) -> int:
+    """Count the number of pattern occurrences"""
+    return (
+        df["class_id"]
```

Comment:
Maybe have the column name be a function parameter too?

Or, instead of passing the entire dataframe, pass only the series?  So:

```python
def count_occurrences(identifier_pattern: str, column: pd.Series) -> int:
   return (
      column
      ...
   )

...

data_count = count_occurrences(data_pattern, df["class_id"])
```

Reply:
Changed function to pass to column